catalog/ddms/catalog-ddms-transformer/src/main/java/org/codice/alliance/ddms/v2/Ddms20InputTransformer.java

```diff
@@ -0,0 +1,762 @@
+
+    // Restriction attributes
+    metacard.setAttribute(getCopyright(ddms));
+    metacard.setAttribute(getIntellectualProperty(ddms));
+    metacard.setAttribute(getPrivacyAct(ddms));
+
+    // Ddms attributes
+    metacard.setAttribute(getInfoCutOff(ddms));
+    metacard.setAttribute(getPosted(ddms));
+    metacard.setAttribute(getGeographicNames(ddms));
+    metacard.setAttribute(getGeographicRegions(ddms));
+    metacard.setAttribute(getBeNumbers(ddms));
+    metacard.setAttribute(getSubtitles(ddms));
+    // TODO: ext.ddms.icid-payload
+    // TODO: ext.ddms.security portion markings
+    // Portion-Markings: Title, Subtitle, description, contacts, related resources, security
+    // TODO: ext.ddms.related-resource not used?
+    // Todo: Not sure what ext.ddms.topic-type is for
+
+    return metacard;
+  }
+
+  private DdmsResource parseDdms(InputStream input) throws CatalogTransformerException {
+    try {
+      XMLStreamReader xmlStreamReader = XMLInputFactory.newFactory().createXMLStreamReader(input);
+      return new Ddms20XmlReader(xmlStreamReader).read();
+    } catch (XMLStreamException e) {
+      throw new CatalogTransformerException("Failed to parse DDMS 2.0", e);
+    }
+  }
+
+  private Attribute getTitle(DdmsResource ddms) {
+    String title = ddms.getTitles().stream().findFirst().map(Title::getValue).orElse(null);
+    return new AttributeImpl(Core.TITLE, title);
+  }
+
+  private Attribute getLocation(DdmsResource ddms) {
+    GeometryCollectionBuilder geometryCollectionBuilder = new GeometryCollectionBuilder();
+    for (GeospatialCoverage geospatialCoverage : ddms.getGeospatialCoverages()) {
+      for (BoundingBox boundingBox : geospatialCoverage.getBoundingBoxes()) {
+        geometryCollectionBuilder.polygon(
+            boundingBox.getWest(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getNorth());
+      }
+
+      for (BoundingGeometry geometry : geospatialCoverage.getBoundingGeometries()) {
+        geometry
+            .getPolygons()
+            .stream()
+            .map(Polygon::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+
+        geometry
+            .getPoints()
+            .stream()
+            .map(Point::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+      }
+    }
+    String location = geometryCollectionBuilder.toString();
+    // can't return blank string or else the solar catalog provider will throw a
+    // NullPointerException
+    if (StringUtils.isBlank(location)) {
+      location = null;
```

Comment:
Can we just not set the attribute?

Reply:
Again, not sure. @beyelerb ?

Reply:
After some discussion, I believe that this is because location is a string, but interpreted as WKT. An empty string would cause the WKT parser (not solr) to throw an exception. Rather than setting a null value, it would probably be best to just not set it in the case that the value is null or is an empty string.

Reply:
Would you recommend the same for the rest of the attributes? Right now, all the attributes that have handlers always get set. If the handlers can't find a value, they just set it to null.

Reply:
Changed only the location handler for now so that it won't be set if the value is null